File: 2 -- Arrays/LinearSearch.cpp
#include <iostream>
using namespace std;

void linearSearch(int arr[],int size,int element){
    bool found=false;

    for(int i=0;i<size;i++){
        if(arr[i]==element){
            found=true;
            cout<< "Element Found At : "<<i;
        }
    }

    if(found==false){
        cout<< "Element not found !";
    }
}

int main(){
    int size;

    cout << "Enter the Size of array : ";
    cin >> size;

    int arr[size];
    for (int i = 0; i < size; i++)
    {
        cout << "Enter the " << i << " Element : ";
        cin >> arr[i];
    }

    int element_toSearch;
    cout<<"Enter element to search : ";
    cin>> element_toSearch;

    linearSearch(arr,size,element_toSearch);
}

File: 2 - Arrays/1D Arr/linearsearch.java
/*THIS CODE WILL SEARCH A SPECIFY NUMBER IN THE ARRAY AND RETURN THE INDEX OF THAT 
 * PARTICULAR INDEX.
 */


import java.util.*;

public class linearsearch{
    int linear(int arr[]){
        Scanner sc=new Scanner(System.in);
        System.out.print("ENTER THE ELEMENT : ");
        int no=sc.nextInt();
        for(int i=0;i<arr.length;i++){
            if(arr[i]==no){
                return i;
            }
        }
        return -1;
    } 

    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.print("ENTER THE SIZE FOR ARRAY : ");
        int size=sc.nextInt();
        int []arr=new int [size];
        System.out.println("ENTER THE ELEMENTS FOR ARRAY : ");
        for(int i=0;i<arr.length;i++){
            arr[i]=sc.nextInt();
        }
        linearsearch l1=new linearsearch();
        int a=l1.linear(arr);
        if(a==-1){
            System.out.print("NO,THIS ELEMNETS DOESN'T EXIST");
        }
        else{
            System.out.print("YES "+"EXIST AT "+a+" INDEX");
        }
    }
}